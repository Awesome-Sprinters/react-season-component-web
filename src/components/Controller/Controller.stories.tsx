import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Controller from './index';

export default {
  component: Controller,
  title: 'Controller',
} as ComponentMeta<typeof Controller>;

/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable-next-line react/function-component-definition */
const Template: ComponentStory<typeof Controller> = args => <Controller {...args} />;

export const Auto = Template.bind({});
Auto.args = {
  type: 'auto',
  width: 500,
  height: 360,
};

export const Spring = Template.bind({});
Spring.args = {
  type: 'spring',
  width: 500,
  height: 360,
};

export const Winter = Template.bind({});
Winter.args = {
  type: 'winter',
  width: 500,
  height: 360,
};
