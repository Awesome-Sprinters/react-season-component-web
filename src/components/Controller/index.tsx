import React from 'react';
import {
  Container,
  Foreground,
  Background,
  ControllerColorType,
  ControllerContentWrapper,
  CommonVariables,
  CustomVariableContainer,
  CardComponentContainer,
  CardComponentHeader,
  AppSlider,
  InputContainer,
  InputProperty,
  InputUnit,
} from './styled';

export interface ControllerProps {
  children: React.ReactNode;
  type: ControllerColorType;
  // 현재 너비
  width: number;
  // 현재 높이
  height: number;
  // 너비 변경 핸들러
  onChangeWidth: (event: Event, value: number | number[]) => void;
  // 높이 변경 핸들러
  onChangeHeight: (event: Event, value: number | number[]) => void;
}

function Controller({
  children = null,
  type,
  width = 200,
  height = 200,
  onChangeWidth,
  onChangeHeight,
}: ControllerProps) {
  return (
    <Container>
      <Foreground type={type} />
      <Background type={type} />
      <ControllerContentWrapper>
        <CustomVariableContainer>
          <CommonVariables>Custom Variables</CommonVariables>
          {children}
        </CustomVariableContainer>
        <CardComponentContainer>
          <CardComponentHeader>Card Component</CardComponentHeader>
          <InputContainer>
            <InputProperty>Width</InputProperty>
            <AppSlider
              size="small"
              defaultValue={width}
              aria-label="Small"
              valueLabelDisplay="auto"
              min={200}
              max={900}
              onChange={onChangeWidth}
            />
            <InputUnit>{width} px</InputUnit>
          </InputContainer>
          <InputContainer>
            <InputProperty>Height</InputProperty>
            <AppSlider
              size="small"
              defaultValue={height}
              aria-label="Small"
              valueLabelDisplay="auto"
              min={200}
              max={600}
              onChange={onChangeHeight}
            />
            <InputUnit>{height} px</InputUnit>
          </InputContainer>
        </CardComponentContainer>
      </ControllerContentWrapper>
    </Container>
  );
}

export default Controller;
