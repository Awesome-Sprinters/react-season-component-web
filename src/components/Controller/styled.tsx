import styled from '@emotion/styled';

export type ControllerColorType =
  | 'spring'
  | 'summer'
  | 'autumn'
  | 'winter'
  | 'morning'
  | 'day'
  | 'evening'
  | 'night'
  | 'sunny'
  | 'cloudy'
  | 'rainy'
  | 'snowy';

export const Container = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  width: 580px;
  height: 780px;
`;

export const Foreground = styled.div<{ type: ControllerColorType }>`
  position: absolute;
  top: 30px;
  left: 0;
  width: 550px;
  background-color: ${({ type }) => {
    switch (type) {
      case 'spring':
        return '#FFF9F3';
      case 'summer':
        return '#FDFEEA';
      case 'autumn':
        return '#FFBB59';
      case 'winter':
        return '#FFF9FD';
      case 'morning':
        return '#FFEEDA';
      case 'day':
        return '#B9D3EE';
      case 'evening':
        return '#4D70C8';
      case 'night':
        return '#7C7A7A';
      case 'sunny':
        return '#D5F8F9';
      case 'cloudy':
        return '#FBE6F8';
      case 'rainy':
        return '#83BDD1';
      default:
        return '#92B7D4';
    }
  }};
  height: 750px;
  border-radius: 50px 50px 0px 300px;
  z-index: 1;
`;

export const Background = styled.div<{ type: ControllerColorType }>`
  position: absolute;
  top: 0;
  right: 0;
  width: 500px;
  height: 575px;
  background-color: ${({ type }) => {
    switch (type) {
      case 'spring':
        return '#FFF3E7';
      case 'summer':
        return '#DAF2EA';
      case 'autumn':
        return '#ffdba8';
      case 'winter':
        return '#FFE7F5';
      case 'morning':
        return '#FFC785';
      case 'day':
        return '#a7d0fc';
      case 'evening':
        return '#94a9db';
      case 'night':
        return '#b9b9b9';
      case 'sunny':
        return '#bbfdff';
      case 'cloudy':
        return '#ffd3f8';
      case 'rainy':
        return '#a9cdda';
      default:
        return '#b9cedf';
    }
  }};
  z-index: 0;
`;
