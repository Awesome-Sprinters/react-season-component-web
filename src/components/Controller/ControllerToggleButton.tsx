import React from 'react';
import VisibilityIcon from '@mui/icons-material/Visibility';
import ToggleButton from '@mui/material/ToggleButton';
import styled from '@emotion/styled';

interface ControllerToggleButtonProps {
  visibility: boolean;
  setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
}

function ControllerToggleButton({ visibility, setVisibility }: ControllerToggleButtonProps) {
  return (
    <StyledToggleButton
      value="check"
      selected={visibility}
      onChange={() => {
        setVisibility(!visibility);
      }}
    >
      <span>Show Controller</span>
      <VisibilityIcon />
    </StyledToggleButton>
  );
}

const StyledToggleButton = styled(ToggleButton)`
  position: fixed;
  bottom: 0;
  right: 0;
  z-index: 1000;
`;

export default ControllerToggleButton;
