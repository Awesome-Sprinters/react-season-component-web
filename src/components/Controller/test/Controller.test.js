import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { renderHook, act } from '@testing-library/react-hooks';

import Controller from '../index';
import useMediaQuery from '../hooks/useMediaQuery';

describe('Controller component test', () => {
  let ControllerComponent;
  let type;

  beforeAll(() => {
    type = 'spring';
    ControllerComponent = (
      <Controller
        type={type}
        width={500}
        height={360}
        onChangeWidth={jest.fn()}
        onChangeHeight={jest.fn()}
      />
    );
  });

  it('Render Controller', () => {
    const component = renderer.create(ControllerComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  //   it('test controller', () => {
  //     const { result } = renderHook(() => useMediaQuery('(max-width: 1300px)'));

  //     expect(result.current).toBe(true);
  //   });
});
