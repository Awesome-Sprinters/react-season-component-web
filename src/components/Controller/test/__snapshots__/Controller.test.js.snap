// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller component test Render Controller 1`] = `
<div
  className="css-by0x3"
>
  <div
    className="css-8r7537"
    type="spring"
  >
    <div
      className="css-0"
    >
      <div
        className="css-hpyoef"
      >
        Custom Variables
      </div>
    </div>
    <div
      className="css-183wyk4"
    >
      <div
        className="css-19f7q00"
      >
        Card Component
      </div>
      <div
        className="css-7s8pi2"
      >
        <div
          className="css-10i6va2"
        >
          Width
        </div>
        <span
          className="MuiSlider-colorPrimary MuiSlider-sizeSmall MuiSlider-root css-10r2c0b-MuiSlider-root"
          defaultValue={500}
          onMouseDown={[Function]}
        >
          <span
            className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            className="MuiSlider-track css-1n40zqk-MuiSlider-track"
            style={
              Object {
                "left": "0%",
                "width": "42.857142857142854%",
              }
            }
          />
          <span
            className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeSmall MuiSlider-thumb css-14gf62f-MuiSlider-thumb"
            data-index={0}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "left": "42.857142857142854%",
                "pointerEvents": undefined,
              }
            }
          >
            <input
              aria-label="Small"
              aria-orientation="horizontal"
              aria-valuemax={900}
              aria-valuemin={200}
              aria-valuenow={500}
              data-index={0}
              disabled={false}
              max={900}
              min={200}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              step={1}
              style={
                Object {
                  "border": 0,
                  "clip": "rect(0 0 0 0)",
                  "direction": "ltr",
                  "height": "100%",
                  "margin": -1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "100%",
                }
              }
              type="range"
              value={500}
            />
            <span
              aria-hidden={true}
              className="MuiSlider-valueLabel css-gdpt3b-MuiSlider-valueLabel"
            >
              <span
                className="MuiSlider-valueLabelCircle"
              >
                <span
                  className="MuiSlider-valueLabelLabel"
                >
                  500
                </span>
              </span>
            </span>
          </span>
        </span>
        <div
          className="css-5pu1uj"
        >
          500
           px
        </div>
      </div>
      <div
        className="css-7s8pi2"
      >
        <div
          className="css-10i6va2"
        >
          Height
        </div>
        <span
          className="MuiSlider-colorPrimary MuiSlider-sizeSmall MuiSlider-root css-10r2c0b-MuiSlider-root"
          defaultValue={360}
          onMouseDown={[Function]}
        >
          <span
            className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            className="MuiSlider-track css-1n40zqk-MuiSlider-track"
            style={
              Object {
                "left": "0%",
                "width": "40%",
              }
            }
          />
          <span
            className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeSmall MuiSlider-thumb css-14gf62f-MuiSlider-thumb"
            data-index={0}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "left": "40%",
                "pointerEvents": undefined,
              }
            }
          >
            <input
              aria-label="Small"
              aria-orientation="horizontal"
              aria-valuemax={600}
              aria-valuemin={200}
              aria-valuenow={360}
              data-index={0}
              disabled={false}
              max={600}
              min={200}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              step={1}
              style={
                Object {
                  "border": 0,
                  "clip": "rect(0 0 0 0)",
                  "direction": "ltr",
                  "height": "100%",
                  "margin": -1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "100%",
                }
              }
              type="range"
              value={360}
            />
            <span
              aria-hidden={true}
              className="MuiSlider-valueLabel css-gdpt3b-MuiSlider-valueLabel"
            >
              <span
                className="MuiSlider-valueLabelCircle"
              >
                <span
                  className="MuiSlider-valueLabelLabel"
                >
                  360
                </span>
              </span>
            </span>
          </span>
        </span>
        <div
          className="css-5pu1uj"
        >
          360
           px
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-17uhyr6"
    type="spring"
  />
</div>
`;
